applications/FluidDynamicsApplication/symbolic_generation/compressible_navier_stokes/compressible_navier_stokes_explicit_cpp_template_with_integration.cpp

```diff
@@ -294,17 +294,18 @@ void CompressibleNavierStokesExplicit<TDim, TNumNodes>::CalculateOnIntegrationPo
     }
 
     if (rVariable == DENSITY_GRADIENT) {
-        const auto& rho_grad = CalculateMidPointDensityGradient();
+        const array_1d<double,3> rho_grad = CalculateMidPointDensityGradient();

```

Comment:
by ref or by value?

Reply:
As we are returning by value a local variable of the `CalculateMidPointDensityGradient` the reference specifier asks for extending the lifetime of the local value to the one of  `rho_grad` (see [here](https://www.learncpp.com/cpp-tutorial/returning-values-by-value-reference-and-address/)). Long story short, I think it makes no real difference in this case (as far as I understand).

Reply:
I agree, should make no difference here

Nice reference